src/pybind/mgr/smb/fs.py

```diff
@@ -108,3 +121,80 @@ def resolve_exists(
                 raise NotADirectoryError(volpath)
         log.debug('Verified that %r exists in %r', volpath, volume)
         return volpath
+
+
+class TTLCache:
+    def __init__(self, maxsize=512, ttl=300):
```

Comment:
mypy should be complaining that this has no type annotations. Even if it isn't let's add them.